src/module_wrap.cc

```diff
@@ -627,6 +630,17 @@ Maybe<const PackageConfig*> GetPackageConfig(Environment* env,
     main_std.assign(std::string(*main_utf8, main_utf8.length()));
   }
 
+  Local<Value> pkg_name;
+  HasName has_name = HasName::No;
+  std::string name_std;
+  if (pkg_json->Get(env->context(), env->name_string()).ToLocal(&pkg_name)) {
+    if (pkg_name->IsString()) {
+      has_name = HasName::Yes;
+    }
+    Utf8Value name_utf8(isolate, pkg_name);
+    name_std.assign(std::string(*name_utf8, name_utf8.length()));
```

Comment:
Shouldn't these lines also be in the `if` block?

Reply:
Good question. I was mirroring the logic for `main_std`.

@guybedford Do you remember why we're still trying to capture those values even if we set `HasMain` to `No` and it failed the `IsString()` check?

Reply:
I think we should be fine with throwing away non-string `name` fields. Will move this assignment into the `if`.